src/identity/identity-wallet.ts

```diff
@@ -400,12 +401,21 @@ export interface IIdentityWallet {
   /**
    *
    * gets profile identity by verifier
-   *
+   * @deprecated The method should not be used. It returns only one profile per verifier, which can potentially restrict business use cases
    * @param {string} verifier -  identifier of the verifier
    * @returns `{Promise<Profile>}`
    */
   getProfileByVerifier(verifier: string): Promise<Profile | undefined>;
 
+  /**
+   * gets profile by verifiers
+   *
+   * @param {string} verifier - verifier to which profile has been shared
+   * @param {string} tag - optional tag to filter profile entry
+   * @returns `{Promise<Profile[]>}`
+   */
+  getProfilesByVerifier(verifier: string, tag?: string): Promise<Profile[]>;
```

Comment:
I recall some time ago we had a discussion with @demonsh and @Kolezhniuk about the name `getProfileByVerifier`. The context was that we were also calling it to create profiles to interact with issuers. This happens for example when we post a credential proposal to the agent.

Would it make sense to take this opportunity to generalize the name of this function? For example:

* `getProfilesByActor`
* `getProfilesByEntity`
* `getProfilesByPeer`

Reply:
Nah, I see this goes far beyond the name of this function, `verifier: string`:

```typescript
export type Profile = {
  id: string;
  nonce: number | string;
  genesisIdentifier: string;
  verifier: string;
  tag?: string; // format of the tag can be defined by client of sdk
};
```